README.md

```diff
@@ -42,138 +42,303 @@ monitoring.
+  that handle administrative tasks. However, actions can theoretically do anything a
+  sensor owner can implement. Some less common (but perfectly acceptable) ideas for
+  actions might be to rotate an antenna, or start streaming data over a socket and only
+  return when the recipient closes the connection.
+
+- *acquisition*: The combination of data and metadata created by an action (though an
+  action does not have to create an acquisition). Metadata is accessible directly
+  though the API, while data is retrievable in an easy-to-use archive format with its
+  associated metadata.
+
+- *admin*: A user account that has full control over the sensor and can create schedule
+  entries and view, modify, or delete any other user's schedule entries or
+  acquisitions.
+
+- *capability*: Available actions, installation specifications (e.g., mobile or
+  stationary), and operational ranges of hardware components (e.g., frequency range of
+  signal analyzer). These values are generally hard-coded by the sensor owner and
+  rarely change.
+
+- *plugin*: A Python package with actions designed to be integrated into scos-sensor.
+
+- *schedule*: The collection of all schedule entries (active and inactive) on the
+  sensor.
+
+- *scheduler*: A thread responsible for executing the schedule. The scheduler reads the
+  schedule at most once a second and consumes all past and present times for each
+  active schedule entry until the schedule is exhausted. The latest task per schedule
+  entry is then added to a priority queue, and the scheduler executes the associated
+  actions and stores/POSTs task results. The scheduler operates in a simple blocking
+  fashion, which significantly simplifies resource deconfliction. When executing the
+  task queue, the scheduler makes a best effort to run each task at its designated
+  time, but the scheduler will not cancel a running task to start another task, even
+  one of higher priority.
+
+- *schedule entry*: Describes a range of scheduler tasks. A schedule entry is at
+  minimum a human readable name and an associated action. Combining different values of
+  *start*, *stop*, *interval*, and *priority* allows for flexible task scheduling. If
+  no start time is given, the first task is scheduled as soon as possible. If no stop
+  time is given, tasks continue to be scheduled until the schedule entry is manually
+  deactivated. Leaving the interval undefined results in a "one-shot" entry, where the
+  scheduler deactivates the entry after a single task is scheduled. One-shot entries
+  can be used with a future start time. If two tasks are scheduled to run at the same
+  time, they will be run in order of *priority*. If two tasks are scheduled to run at
+  the same time and have the same *priority*, execution order is
+  implementation-dependent (undefined).
+
+- *signal*: Django notifications library used to allow scos-sensor to receive
```

Comment:
action completion is just one example right? Isn't a broader framework for event driven programming/architecture?